client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -343,4 +344,9 @@
     * Executes a remote script passing a set of named parameters
     */
    <T> T execute(String scriptName, Map<String, ?> params);
+
+   /**
+    * Returns segment ownership for this cache
+    */
+   Map<SocketAddress, Set<Integer>> getSegmentsByServer();
```

Comment:
Yeah, I understand, but then I'd probably prefer to encapsulate the return type into an interface like RemoteSegmentMapping instead of Map<SocketAddress, Set<Integer>> (same for retrieveEntries).